code/datums/achievements/_awards.dm

```diff
@@ -217,3 +233,114 @@
 		holder.data[type] = text2num(get_unlocked_load.item[1]) || 0
 		holder.original_cached_data[type] = 0
 	qdel(get_unlocked_load)
+
+/**
+ * A subtype of score linked to a schema table containing objects the player has caught, made, found or otherwise achieved.
+ * The value of the score should equal to the length of objects that have been achieved.
+ * It also has an unique tab in the UI that lets you review the progress.
+ */
+/datum/award/score/progress
+	///The id of table we're looking for, the one containing the progress objects.
+	var/table_id = ""
```

Comment:
rather then being a variable, make a getter proc that returns the table to use, and never reuse its result.
we need to avoid admins being able to insert into any random db table

Reply:
I've made the two vars uneditable by VV, is that good for you?

Reply:
no, vv blocks can be bypassed. normally this is.... ok but if the table is arbitrary it is not

Reply:
Fine, I'll replace them with getters later.